packages/paste-core/components/chat-log/src/ChatMessageMeta.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+
+import {MessageVariantContext} from './MessageVariantContext';
+import type {ChatMessageMetaProps} from './types';
+
+const ChatMessageMeta = React.forwardRef<HTMLDivElement, ChatMessageMetaProps>(
+  ({children, element = 'CHAT_MESSAGE_META', ...props}, ref) => {
+    const variant = React.useContext(MessageVariantContext);
+
+    return (
+      <Box
+        ref={ref}
+        element={element}
+        display="flex"
+        justifyContent={Array.isArray(children) ? 'space-between' : 'flex-end'}
+        textAlign={!Array.isArray(children) ? 'right' : undefined}
```

Comment:
Nit: Minor optimisation, you could call Array.isArray(children) once outside of the render block to save doing it twice

Reply:
Minor improvement on `Array.isArray` is to use the React top-level API `React.Children.count(children)` https://reactjs.org/docs/react-api.html#reactchildrencount